wikiHandler.js

```diff
@@ -86,10 +93,46 @@ function formatPrice(price) {
     }
 }
 
+function parsePage(page, server) {
+    const $ = cheerio.load(page);
+    const auc_data = $(`#auc_${server} .eoTable3 td`).contents();
+    let price_data = {};
+    if (auc_data !== undefined && auc_data[0] !== undefined) {
+        price_data[30] = auc_data[0].data.trim();
+    }
+    if (auc_data !== undefined && auc_data[1] !== undefined) {
+        price_data[90] = auc_data[1].data.trim();
+    }
+    return price_data;
+}
+
 async function getWikiPricing(item_url, server) {
     // Translate server name to "capitalized" form, eg: GREEN -> Green
     server = server[0] + server.slice(1).toLowerCase();
 
+    //check cache before fetching
+    cache.get(item_url, (err, cached_data) => {
+        if (err) console.error(err);
+        //use cached data if available
+        if (cached_data) {
+            console.log('cached worked for ', item_url)
+            parsePage(cached_data, server)
```

Comment:
So we're storing the raw page response in cache? I guess this is fine, but it'd be much smaller to cache the actual result we want, right? And then we'd only have to run parsePage once before storing, not every time we fetch.

Reply:
100%.  Was thinking that this morning.  Now it's much more proper.